test/integration-ebpf/src/tc_name_limit.rs

```diff
@@ -0,0 +1,28 @@
+#![no_std]
+#![no_main]
+
+use aya_bpf::{macros::classifier, programs::TcContext};
+
+// This macro generates a function with arbitrary name
+macro_rules! generate_ebpf_function {
+    ($fn_name:ident) => {
+        #[classifier]
+        pub fn $fn_name(_ctx: TcContext) -> i32 {
+            0
+        }
+    };
+}
+
+/*
+Generating a function with a 256-byte-long name (all 'a's) to be used as
+the ebpf program. This name must match the name passed to userspace side.
+256 is the maximum length allowed by the kernel:
+https://github.com/torvalds/linux/blob/02aee814/net/sched/cls_bpf.c#L28
+*/
+generate_ebpf_function!(aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa);
```

Comment:
what happens if the program name is 257 bytes long?

Reply:
@tamird Sorry for the delay
This is the error you get when the name is more that 256 bytes:
```
Error: netlink error while attaching ebpf program to tc

Caused by:
    Invalid argument (os error 22)
```

Reply:
That's a good idea, Macro is probably more useful now.

Reply:
@tamird An integration test was added to make sure programs with a name longer than 256 bytes will fail to attach.